pyperformance/_pip.py

```diff
@@ -149,10 +149,16 @@ def install_requirements(reqs, *extra,
     args = []
     if upgrade:
         args.append('-U')  # --upgrade
-    for reqs in [reqs, *extra]:
-        if os.path.exists(reqs):
-            args.append('-r')  # --requirement
-        args.append(reqs)
+    for req in [reqs, *extra]:
+        if os.path.isfile(req):
+            name = os.path.basename(req)
+            if name == "setup.py":
+                req = os.path.dirname(req)
```

Comment:
The reason we use the dirname isn't obvious, so it may be worth adding a comment here indicating pip's limitations.